AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/RecyclerViewMrecAdActivity.java

```diff
@@ -0,0 +1,250 @@
+package com.applovin.enterprise.apps.demoapp.ads.max.mrecs;
+
+import android.app.Activity;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import android.widget.TextView;
+
+import com.applovin.enterprise.apps.demoapp.R;
+import com.applovin.mediation.MaxAd;
+import com.applovin.mediation.MaxAdFormat;
+import com.applovin.mediation.MaxAdViewAdListener;
+import com.applovin.mediation.MaxError;
+import com.applovin.mediation.ads.MaxAdView;
+import com.applovin.sdk.AppLovinSdkUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+public class RecyclerViewMrecAdActivity
+        extends AppCompatActivity
+{
+    private final int                   AD_VIEW_COUNT = 5;
+    private final int                   AD_INTERVAL   = 10;
+    private final ArrayList<String>     sampleData    = new ArrayList<>( Arrays.asList( "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".split( "" ) ) );
+    private final List<MaxAdView>       adViews       = new ArrayList<>( AD_VIEW_COUNT );
+    private       CustomRecyclerAdapter adapter;
+
+    @Override
+    protected void onCreate(@Nullable final Bundle savedInstanceState)
+    {
+        super.onCreate( savedInstanceState );
+        setContentView( R.layout.activity_mrec_recycler_view );
+        setTitle( R.string.activity_recycler_view_mrecs );
+
+        adapter = new CustomRecyclerAdapter( this, sampleData );
+
+        // Configure recycler view
+        RecyclerView recyclerView = findViewById( R.id.mrec_recycler_view );
+        recyclerView.setLayoutManager( new LinearLayoutManager( this ) );
+        recyclerView.setAdapter( adapter );
+
+        configureAdViews();
+    }
+
+    private void configureAdViews()
+    {
+        // Insert rows at each interval to be used to display an ad
+        for ( int i = 0; i < sampleData.size(); i += AD_INTERVAL )
+        {
+            sampleData.add( i, "" );
+            adapter.notifyDataSetChanged();
+        }
```

Comment:
We are actually updating the list as we are iterating through it. I don't think the ads would be added at the desired positions. Since the list size changes as we add new items.

Reply:
Could you clarify how the ads would be in different positions? In this example, ads are being added at specific indices 0, 10, 20, 30. `sampleData.size()` gets evaluated at each iteration so it gets incremented each time we add new items. And because of the for loop condition, we only ever add an item if the list is big enough. 

I believe calling `adapter.notifyDataSetChanged()` updates the position of each view in the recycler view to the correct result.

Reply:
Ah, my bad. It felt like we'd be drifting by one on each iteration as we are adding new items, but looks like we'll always have 9 items between ads.